### DISCUSSION TITLE
Serve images in next-gen formats

### DISCUSSION THREAD

**Discussion by Mryuksel101** [CaiJimmy/hugo-theme-stack#652] [Q&A] [answered]
![Ekran Görüntüsü (631)](https://user-images.githubusercontent.com/99680265/182148559-f89b11d8-2964-481c-a957-3c533ccd9661.png)

How can I do that? I am using github pages not netlify

**Comment by rmdhnreza** [comment, marked as answer]
1. You can convert images manually to WebP, for example with [squoosh.app](https://squoosh.app)
2. Automatically using `render-image` hooks.
Edit `layouts/_default/_markup/render-image.html` and add `webp` to line 21-22

https://github.com/CaiJimmy/hugo-theme-stack/blob/5869a7ab83aaff19393433f0547d1db40ca437da/layouts/_default/_markup/render-image.html#L21-L22

like this

```
...

 {{- $small := $image.Resize `webp 480x` -}} 
 {{- $big := $image.Resize `webp 1024x` -}}

...

```


make sure you enable `imageProcessing` on config.yaml

https://github.com/CaiJimmy/hugo-theme-stack/blob/5869a7ab83aaff19393433f0547d1db40ca437da/config.yaml#L139-L143

AFAIK, Hugo still not support AVIF for target format, so WebP is the best option for now.

**Reply by Mryuksel101** [reply]
![image](https://user-images.githubusercontent.com/99680265/182619523-4a14d07e-bc14-43c0-921c-0a3301fc5af4.png)

hi, i solved that problem. Thank you so much.
But my picture doesn't look good and I don't know what to do

**Reply by rmdhnreza** [reply]
Did you mean blurry? [Default quality](https://gohugo.io/content-management/image-processing/#quality) for image processing is 75, you can add q100 after `webp`
